nightlies/cron_wrapper

```diff
@@ -37,6 +46,10 @@ if [ $code != 0 ] ; then
 else
         # normal exit: just log stdout
         DATE=$(date)
-        echo -n "$DATE: " >> $LOG
-        cat $STDOUT  >> $LOG
+        # if flock failed, define a new $LOG and keep runing
+        flock 200 || $LOG
```

Comment:
however, if you failed to get the lock above, you probably want to continue using that fallback lock file for the rest of the job, otherwise messages will exist in both places, which would be bad.  Also also, you probably want the fallback log file name to be unique per-job (simplest method, adding $$ to the name).